app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -91,6 +89,18 @@ def unit_id_field
         "#{self.name.demodulize.underscore}_id"
       end
 
+      def copy_repository_associations(source_repo, dest_repo)
```

Comment:
what is the difference in the Intent of this method vs sync repository associations in the same class

Reply:
Sync repository associations takes a list of ids or uuids and sets the associations to those ids.  copy_repository_associations copies them in the db from on repo to another.